client/components/Editor/plugins/keymap.ts

```diff
@@ -18,14 +18,37 @@ import {
 } from 'prosemirror-commands';
 import { wrapInList, splitListItem, liftListItem, sinkListItem } from 'prosemirror-schema-list';
 import { undo, redo } from 'prosemirror-history';
+import { Schema } from 'prosemirror-model';
 import { undoInputRule } from 'prosemirror-inputrules';
 import { keymap } from 'prosemirror-keymap';
+import { Selection, EditorState, Command, Transaction } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
 import { mathBackspaceCmd } from '@benrbray/prosemirror-math';
 
 import { splitBlockPreservingAttrs } from '../commands';
 
 const mac = typeof navigator !== 'undefined' ? /Mac/.test(navigator.platform) : false;
 
+type Direction = 'left' | 'up' | 'down' | 'right';
+type DispatchFn = (tr: Transaction) => void;
+const arrowHandler =
+	(dir: Direction): Command =>
+	(state: EditorState, dispatch?: DispatchFn, view?: EditorView): boolean => {
+		if (state.selection.empty && view && view.endOfTextblock(dir)) {
+			const side = dir === 'left' || dir === 'up' ? -1 : 1;
+			const $head = state.selection.$head;
+			const nextPos = Selection.near(
+				state.doc.resolve(side > 0 ? $head.after() : $head.before()),
+				side,
+			);
+			if (nextPos.$head && nextPos.$head.parent.type.name === 'code_block') {
+				dispatch?.(state.tr.setSelection(nextPos));
```

Comment:
Just learned that this conditional calling is possible. I'd seen it for conditional access of properties, but not considered that it's a terse `if` here, too.